src/__tests__/__snapshots__/index.js.snap

```diff
@@ -106,11 +107,27 @@ const red = macro('noop');
       ↓ ↓ ↓ ↓ ↓ ↓
 
 const macro = require('./fixtures/keep-imports.macro');
-
 const red = macro('noop');
 
 `;
 
+exports[`macros optionally keep imports in combination with babel-preset-env (#80): optionally keep imports in combination with babel-preset-env (#80) 1`] = `
+
+import macro from './fixtures/keep-imports.macro'
+const red = macro('noop')
+
+      ↓ ↓ ↓ ↓ ↓ ↓
+
+"use strict";
+
+var _keepImports2 = _interopRequireDefault(_keepImports);
```

Comment:
This looks wrong. That `_keepImports` variable is not defined.

Reply:
You're right! Is that the reason why macros fail with env plugin? I'll take a look...

Reply:
It was indeed related. First the macro is executed on `import macro from 'package/macro`. That's ok.

If import isn't removed, then `babel/env` transforms it into `require` and macro runs again against `const _macro = require('package/macro')`. This caused the problem with unexisting binding, because `_macro` isn't used, but rather `_macro2`. So, after I fixed the typeerror, `applyMacros` returned a null, because there was no reference and therefore the import was removed... 

To keep imports correctly, I have to run macro with empty references in such cases, because I need to get either `null` or `{ keepMacros: true}` as a result from macro.